dateconv/dateconv.go

```diff
@@ -47,6 +47,30 @@ func (b BSDate) After(u BSDate) bool {
 	return dateA > dateB
 }
 
+// CheckBounds checks if the conversion is possible. Raises an error is an out
+// of bound input is provided. Date after 2000 Baisakh 1 should be provided.
+func (b BSDate) CheckBounds() error {
```

Comment:
Please add some tests for these, would rather merge these in with some (trivial) tests even though they are pretty simple.